fix: handle case where email is deleted after sending, user hits unsub

donbowman opened the pull request:
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 4.0
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 


#### Description:


#### Steps to test this PR:
Send a segment email with an unsubscribe link.
Delete the email.
Click the unsubscribe link. You will get a 500 error. After this, you will get the generic preferences page.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10423"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

jos0405 commented:
Hi,
I tested this out and works as it supposed to.
After deleting the original email the unsubscribe link still works:

![image](https://user-images.githubusercontent.com/4246909/133208908-5988fc25-f382-4933-9f9d-696a0a5b42fb.png)

kuzmany commented:
I see some files what are not related to this PR.
Could you rebase it related to 4.0 branch?

kuzmany reviewed (dismissed):
Works as expected. Thanks!

dennisameling commented:
@donbowman this PR reduces our test coverage. We require our test coverage to either stay the same or increase when PRs are created. Would you mind adding a test? I just added the skeleton for that in https://github.com/mautic/mautic/pull/10423/commits/80274235e10103d1a6da99c37716f827baecd935 with several TODOs. Let us know if you need help with this!

donbowman commented:
> @donbowman this PR reduces our test coverage. We require our test coverage to either stay the same or increase when PRs are created. Would you mind adding a test? I just added the skeleton for that in [8027423](https://github.com/mautic/mautic/commit/80274235e10103d1a6da99c37716f827baecd935) with several TODOs. Let us know if you need help with this!

ok, I will try.
i don't have any php knowledge, and not really looking to gain much at this stage of its lifecycle, but i'll see what i can do. Given there was zero tests here it was very difficult to figure out how to build a framework to correct a missing `if`.

donbowman committed:
fix: handle case where email is deleted after sending, user hits unsub

Create an email. Send to segment. Delete. User opens email afterwards,
selects unsub. Gets 500 error.

donbowman commented:
so i've rebased this.
i've taken a crack at the tests, but am stuck. Any advice?

specifically, is there an example test case somewhere that gets an id hash for an email?
i created the segment, the email, added the list, but i cannot determine how one gets the id hash for the unsub.

kuzmany commented:
@jos0405 I think this was already fixed but another PRs or code.
